coursedata/yaml_transformer.py

```diff
@@ -18,9 +19,11 @@ def transform_level_defaults(transformations):
         if level == old_level:
           transformed_dict[new_level] = transformed_dict[old_level]
           del transformed_dict[old_level]
+          file_transformed = True
 
-    with open(output_path + yaml_filesname_without_path, 'w') as f:
-      f.write(utils.dump_yaml_rt(transformed_dict))
+    if file_transformed:
```

Comment:
You commented on the reason of this guard on line 49. Do you want to comment it here as well?